File: scripts/check_counts.js
const fs = require('fs');
const path = require('path');

// Read dotenv
const envContent = fs.readFileSync(path.join(__dirname, '.env'), 'utf8');
const env = {};
envContent.split('\n').forEach(line => {
  const parts = line.split('=');
  if (parts.length >= 2) {
    const k = parts[0].trim();
    const v = parts.slice(1).join('=').trim().replace(/^['"]|['"]$/g, '');
    env[k] = v;
  }
});

const url = env.SUPABASE_URL;
const key = env.SUPABASE_SERVICE_ROLE_KEY;

const headers = {
  "apikey": key,
  "Authorization": `Bearer ${key}`,
  "Content-Type": "application/json"
};

const tables = ["visits", "summaries", "specialty_classification", "product_calls", "coaching_days"];

async function main() {
  for (const t of tables) {
    try {
      const fetch = (await import('node-fetch')).default;
      // Note: we can use standard fetch since Node 18 has global fetch!
      // Let's use global fetch first, fallback to node-fetch if needed.
      const fn = typeof fetch === 'function' ? fetch : global.fetch;
      const res = await fn(`${url}/rest/v1/${t}?select=count&period=eq.July 2026`, { headers });
      if (res.ok) {
        const text = await res.text();
        console.log(`Table ${t} for 'July 2026': ${text}`);
      } else {
        console.log(`Failed to fetch ${t}: ${res.status} ${await res.text()}`);
      }
    } catch (err) {
      console.log(`Error fetching ${t}: ${err.message}`);
    }
  }

  // Check RPC
  try {
    const fn = global.fetch;
    const payload = {
      "p_period": "July 2026",
      "p_codes": ["7535"],
      "p_is_admin": true,
      "p_is_manager": true,
      "p_start_date": "2026-07-01",
      "p_end_date": "2026-07-31"
    };
    const res = await fn(`${url}/rest/v1/rpc/get_dashboard_data`, {
      method: 'POST',
      headers,
      body: JSON.stringify(payload)
    });
    if (res.ok) {
      const data = await res.json();
      console.log("\nRPC get_dashboard_data keys returned:", Object.keys(data));
      for (const [k, v] of Object.entries(data)) {
        console.log(`  ${k}: ${Array.isArray(v) ? v.length : typeof v}`);
      }
    } else {
      console.log(`\nFailed to call RPC get_dashboard_data: ${res.status} ${await res.text()}`);
    }
  } catch (err) {
    console.log(`Error calling RPC: ${err.message}`);
  }
}

main();

File: scripts/check_counts.py
import os
import requests

# Read .env manually
env = {}
env_path = "e:\\ExcellenceSystem\\scripts\\.env"
if os.path.exists(env_path):
    with open(env_path, "r", encoding="utf-8") as f:
        for line in f:
            line = line.strip()
            if line and not line.startswith("#") and "=" in line:
                k, v = line.split("=", 1)
                env[k.strip()] = v.strip().strip("'\"")

url = env.get("SUPABASE_URL")
key = env.get("SUPABASE_SERVICE_ROLE_KEY")

headers = {
    "apikey": key,
    "Authorization": f"Bearer {key}",
    "Content-Type": "application/json"
}

print("Fetching counts...")
h_res = requests.get(f"{url}/rest/v1/hierarchy?select=count", headers=headers)
s_res = requests.get(f"{url}/rest/v1/summaries?select=count&period=eq.July 2026", headers=headers)
v_res = requests.get(f"{url}/rest/v1/visits?select=count&period=eq.July 2026", headers=headers)

print("Hierarchy row count:", h_res.json() if h_res.ok else h_res.text)
print("Summaries (July 2026) row count:", s_res.json() if s_res.ok else s_res.text)
print("Visits (July 2026) row count:", v_res.json() if v_res.ok else v_res.text)

print("\nFetching data...")
# Fetch samples
h_data = requests.get(f"{url}/rest/v1/hierarchy?select=employee_code,employee_name&limit=10", headers=headers).json()
s_data = requests.get(f"{url}/rest/v1/summaries?select=employee_code,user_name&period=eq.July 2026&limit=10", headers=headers).json()

print("\nHierarchy codes sample:", [r.get("employee_code") for r in h_data])
print("Summaries codes sample:", [r.get("employee_code") for r in s_data])

# Check intersection
h_all = requests.get(f"{url}/rest/v1/hierarchy?select=employee_code", headers=headers).json()
s_all = requests.get(f"{url}/rest/v1/summaries?select=employee_code&period=eq.July 2026", headers=headers).json()

h_codes = set(r.get("employee_code") for r in h_all if isinstance(r, dict) and r.get("employee_code"))
s_codes = set(r.get("employee_code") for r in s_all if isinstance(r, dict) and r.get("employee_code"))

print(f"\nTotal unique codes in hierarchy: {len(h_codes)}")
print(f"Total unique codes in summaries (July 2026): {len(s_codes)}")
print(f"Intersection of codes: {len(h_codes.intersection(s_codes))}")
